alma/controllers/hook/DisplayPaymentHookController.php

```diff
@@ -74,78 +82,86 @@ public function run($params)
-                    'desc' => sprintf(SettingsCustomFields::getPaymentButtonDescriptionDeferredByLang($idLang), $duration),
+                    'installmentText' => $this->getInstallmentText($plans, $idLang),
+                    'deferred_trigger_limit_days' => $feePlans->$key->deferred_trigger_limit_days,
+                    'isDeferred' => $isDeferred,
+                    'text' => sprintf(SettingsCustomFields::getPaymentButtonTitleByLang($idLang), $installment),
+                    'desc' => sprintf(SettingsCustomFields::getPaymentButtonDescriptionByLang($idLang), $installment),
                     'creditInfo' => $creditInfo,
-                    ];
-                    $paymentOptions[$key] = $paymentOption;
-                    $sortOptions[$key] = $feePlans->$key->order;
-                }
-            } else {
-                if ($n != 1) {
-                    if (!$plan->isEligible && $feePlans->$key->enabled && Settings::showDisabledButton()) {
-                        $disabled = true;
-                        $plans = null;
-                    } elseif (!$plan->isEligible) {
-                        continue;
-                    }
-                    $logo = $this->getAlmaLogo(false, $n);
-                    $paymentOption = [
-                        'link' => $this->context->link->getModuleLink(
-                            $this->module->name,
-                            'payment',
-                            ['key' => $key],
-                            true
-                        ),
-                        'disabled' => $disabled,
-                        'plans' => $plans,
-                        'pnx' => $n,
-                        'logo' => $logo,
-                        'isDeferred' => false,
-                        'text' => sprintf(SettingsCustomFields::getPaymentButtonTitleByLang($idLang), $n),
-                        'desc' => sprintf(SettingsCustomFields::getPaymentButtonDescriptionByLang($idLang), $n),
-                        'creditInfo' => $creditInfo,
-                    ];
-                    $paymentOptions[$key] = $paymentOption;
-                    $sortOptions[$key] = $feePlans->$key->order;
+                ];
+                if ($isDeferred) {
+                    $paymentOption['duration'] = $duration;
+                    $paymentOption['key'] = $key;
+                    $paymentOption['text'] = sprintf(SettingsCustomFields::getPaymentButtonTitleDeferredByLang($idLang), $duration);
+                    // phpcs:ignore
+                    $paymentOption['desc'] = sprintf(SettingsCustomFields::getPaymentButtonDescriptionDeferredByLang($idLang), $duration);
                 }
+                $paymentOptions[$key] = $paymentOption;
+                $sortOptions[$key] = $feePlans->$key->order;
             }
         }
 
         asort($sortOptions);
         $payment = [];
-        foreach ($sortOptions as $key => $option) {
+        $keyPlans = array_keys($sortOptions);
```

Comment:
you can inline this into the `foreach` to avoid memory consomption due to variable intialization because `$keyPlans` is not used anywhere else

Reply:
Done